diamond-api/src/render.rs

```diff
@@ -464,7 +463,7 @@ mod tests {
     }
 
     #[test]
-    fn url_deserialize_time_relative() -> Result<(), Error> {
+    fn url_deserialize_time_relative() -> Result<(), Box<dyn error::Error>> {
```

Comment:
Code is inconsistent. In some files it is `error::Error` in other it is just `Error`.

Reply:
in some Error was imported from another place

Reply:
1. Do not use actix_web::Error outside of controllers (controller modules).
2. Use actix_web::Error exclusively inside of controllers (controller modules).
3. Functions returning std::io::Result look like independent and can be extracted into their own modules.
3(a). alternatively, import io like `use std::io` and use `io::Result` and `io::Error`.